modules/engine/src/transform/texture-transform.js

```diff
@@ -211,14 +213,16 @@ export default class TextureTransform {
   }
 
   _updateBinding(binding, opts) {
-    const {sourceTextures, targetTexture} = opts;
+    const {sourceTextures, targetTexture, sourceBuffers} = opts;
     if (!binding) {
       binding = {
+        sourceBuffers: {},
         sourceTextures: {},
         targetTexture: null
       };
     }
     Object.assign(binding.sourceTextures, sourceTextures);
+    Object.assign(binding.sourceBuffers, sourceBuffers);
```

Comment:
I'm concerned about all these `Object.assign` allocations. How frequently is `_updateBinding` called?

Reply:
These object sizes are small, (not more than 5 fields for typical use case), and `assign` is executed during initialization and when a binding (Buffer or Texture) is updated. Most of the times this is done once for multiple draw calls.